python_modules/libraries/dagster-k8s/dagster_k8s/client.py

```diff
@@ -730,14 +730,25 @@ def _has_container_logs(self, container_status):
 
         return False
 
-    def get_pod_debug_info(self, pod_name, namespace, container_name: Optional[str] = None) -> str:
-        pods = self.core_api.list_namespaced_pod(
-            namespace=namespace, field_selector="metadata.name=%s" % pod_name
-        ).items
-        pod = pods[0] if pods else None
+    def get_pod_debug_info(
+        self,
+        pod_name,
+        namespace,
+        container_name: Optional[str] = None,
+        pod: Optional[kubernetes.client.V1Pod] = None,  # the already fetched pod
+    ) -> str:
+        if pod is None:
+            pods = self.core_api.list_namespaced_pod(
+                namespace=namespace, field_selector="metadata.name=%s" % pod_name
+            ).items
+            pod = pods[0] if pods else None
```

Comment:
guaranteed to be only of length one? should we do a `check`?

Reply:
ya the `=` field selector should only match the one pod, cant have name conflicts within a namespace